File: modules/playground/src/async/http_request_app.dart
import 'package:angular2/html.dart';

import 'package:angular2/angular2.dart' show Component;

@Component(
  selector: 'http-request-app',
  template: '''
    <span class='val'>{{val}}</span>
    <button class='action' (click)="httpRequest()">Http Request</button>''')
class HttpRequestApp {
  String val = 'placeholder';

  ScrollAreaComponent() {
  }
  
  httpRequest() {
    HttpRequest.getString('slowslowslowdata.json')
        .then((String fileContents) {
          this.val = fileContents;
        });
      }
}

File: modules/playground/src/async/http_request_app.ts
import {Component} from 'angular2/core';
import {HTTP_PROVIDERS, Http, Response} from 'angular2/http';

@Component({
  selector: 'http-request-app',
  template: `
    <span class='val'>{{val}}</span>
    <button class='action' (click)="httpRequest()">Http Request</button>
  `,
  providers: [HTTP_PROVIDERS]
})
export class HttpRequestApp {
  http: Http;
  val: string = 'placeholder';

  constructor(http: Http) { this.http = http; }

  httpRequest(): void {
    this.http.get('slowslowslowdata.json').subscribe((res: Response) => { this.val = res.text(); });
  }
}
